packages/widgets/src/lib/components/InferenceWidget/widgets/ConversationalWidget/ConversationalWidget.svelte

```diff
@@ -115,34 +145,41 @@
 		}
 	}
 
-	function isValidOutput(arg: any): arg is Response {
-		return (
-			arg && Array.isArray(arg?.conversation?.generated_responses) && Array.isArray(arg?.conversation?.past_user_inputs)
-		);
-	}
-
-	function parseOutput(body: unknown): {
-		conversation: Conversation;
+	function parseOutput(
+		body: unknown,
+		chat: Message[]
+	): {
+		chat: Message[];
 		output: Output;
 	} {
-		if (isValidOutput(body)) {
-			const conversation = body.conversation;
-			const pastUserInputs = conversation.past_user_inputs;
-			const generatedResponses = conversation.generated_responses;
-			const output = pastUserInputs
-				.filter(
-					(x, i) =>
-						x !== null && x !== undefined && generatedResponses[i] !== null && generatedResponses[i] !== undefined
-				)
-				.map((x, i) => ({
-					input: x ?? "",
-					response: generatedResponses[i] ?? "",
-				}));
-			return { conversation, output };
+		if (Array.isArray(body) && body.length) {
+			const text = body[0]?.generated_text ?? "";
+
+			if (!text.length) {
+				throw new Error("Model did not generate a response.");
+			}
+
+			const chatWithOutput = [...chat, { role: "assistant", content: text }];
+
+			const output: Output = Array.from({ length: Math.floor(chatWithOutput.length / 2) }, (_, index) => ({
+				input: chatWithOutput[2 * index].content,
+				response: chatWithOutput[2 * index + 1].content,
+			}));
```

Comment:
We probably don't need the `output` variable - we can rely on `messages` entirely

Requires updating the display logic inside `WidgetOutputConvo`

Reply:
Addressed in https://github.com/huggingface/huggingface.js/pull/457/commits/05b333ab13da8aba6ad478707fa8fbc8b5454c9f :)